src/gballoc.c

```diff
@@ -266,7 +267,7 @@ void* gballoc_realloc(void* ptr, size_t size)
                 }
 
                 /* Codes_SRS_GBALLOC_01_007: [If realloc is successful, gballoc_realloc shall also increment the total memory used value tracked by this module.] */
-                totalSize += size;
+                totalSize = safe_add_size_t(totalSize, size);
```

Comment:
Should check for overflow.

Reply:
not needed. This is only for malloc stats update